telescope/ERQTelescopeTrackFinder.cxx

```diff
@@ -156,54 +169,66 @@ void ERQTelescopeTrackFinder::Exec(Option_t* opt) {
-          fTargetX = trackFromMWPC->GetTargetX();
-          fTargetY = trackFromMWPC->GetTargetY();
-          fTargetZ = trackFromMWPC->GetTargetZ();
+        // Calc unknown coordinated using condition: target, hit on first station(closest) and
+        // hit on second station lie on line :
+        // {x1, y1, z1} = {fTargetX, fTargetY, fTargetZ} + k * ({x1, y1, z1} - {fTargetX, fTargetY, fTargetZ}).
+        const bool xStationIsClosest = fQTelescopeSetup->GetStripGlobalZ(xDigiBranchName, xStripNb)
+                                       < fQTelescopeSetup->GetStripGlobalZ(yDigiBranchName, yStripNb);
+        // We know all about z coordinate, so 
+        const double z1 = xStationIsClosest 
+                          ? fQTelescopeSetup->GetStripGlobalZ(xDigiBranchName, xStripNb)
+                          : fQTelescopeSetup->GetStripGlobalZ(yDigiBranchName, yStripNb);
+        const double z2 =  xStationIsClosest 
+                          ? fQTelescopeSetup->GetStripGlobalZ(yDigiBranchName, yStripNb)
+                          : fQTelescopeSetup->GetStripGlobalZ(xDigiBranchName, xStripNb);
+        assert(z1 != fTargetZ);
+        const double k = (z2 - fTargetZ) / (z1 - fTargetZ);
+        double x1 = 0., x2 = 0., y1 = 0., y2 = 0.;
+        if (xStationIsClosest) { // find y1, x2 from equation
+          x1 = fQTelescopeSetup->GetStripGlobalX(xDigiBranchName, xStripNb);
+          y2 = fQTelescopeSetup->GetStripGlobalY(yDigiBranchName, yStripNb);
+          y1 = (-1./k)*((1. - k)*fTargetY - y2);
+          x2 = (1. - k)*fTargetX + k*x1;
+        } else { // find x1, y2 from equation
+          x2 = fQTelescopeSetup->GetStripGlobalX(xDigiBranchName, xStripNb);
+          y1 = fQTelescopeSetup->GetStripGlobalY(yDigiBranchName, yStripNb);
+          x1 = (-1./k)*((1. - k)*fTargetX - x2);
+          y2 = (1. - k)*fTargetY + k*y1;
         }
+        const auto xQTeleGlobHit = (x1 + x2) / 2.;
+        const auto yQTeleGlobHit = (y1 + y2) / 2.;
+        const auto zQTeleGlobHit = (z1 + z2) / 2.;
+        const auto xQTeleLocalHit = fQTelescopeSetup->GetStripLocalX(xDigiBranchName, xStripNb);
+        const auto yQTeleLocalHit = fQTelescopeSetup->GetStripLocalY(yDigiBranchName, yStripNb);
+        const auto zQTeleLocalHit = (fQTelescopeSetup->GetStripLocalZ(xDigiBranchName, xStripNb)
+                                     + fQTelescopeSetup->GetStripLocalZ(yDigiBranchName, yStripNb))/2.;
+        LOG(DEBUG) << " Local hit X Y Z " << xQTeleLocalHit << " " << yQTeleLocalHit 
+                   << " " << zQTeleLocalHit << FairLogger::endl;
+        LOG(DEBUG) << " Global hit X Y Z " << xQTeleGlobHit << " " << yQTeleGlobHit 
+                   << " " << zQTeleGlobHit << FairLogger::endl;
+        
         //Double_t sumEdep = yStrip->GetEdep();        
-        Double_t sumEdep = xStrip->GetEdep();        
-        ERQTelescopeTrack *track = AddTrack(fTargetX, fTargetY, fTargetZ, 
+        const Double_t sumEdep = xStrip->GetEdep();        
```

Comment:
It must be not an xStrip to get energy but the strip from the farthest station from the target

Reply:
And since we don't use sumEdep actually in PID process in the current PR it is becoming obsolete